include/stdx/type_traits.hpp

```diff
@@ -87,5 +87,30 @@ template <typename T> struct type_identity {
     using type = T;
 };
 template <typename T> using type_identity_t = typename type_identity<T>::type;
+
+template <typename...> struct type_list;
+template <auto...> struct value_list;
+
+template <typename L> struct for_each_t {
+    static_assert(always_false_v<L>,
+                  "for_each_type (or for_each_value) must be called with a "
+                  "type list (or value list)");
+};
+
+template <template <typename...> typename L, typename... Ts>
+struct for_each_t<L<Ts...>> {
+    template <typename F> constexpr auto operator()(F &&f) const {
+        (f.template operator()<Ts>(), ...);
+    }
+};
+template <template <auto...> typename L, auto... Vs>
+struct for_each_t<L<Vs...>> {
+    template <typename F> constexpr auto operator()(F &&f) const {
+        (f.template operator()<Vs>(), ...);
+    }
+};
+
+template <typename L> constexpr static auto for_each_type = for_each_t<L>{};
```

Comment:
question: since there is no distinction between the two calls (you can call `for_each_type` with values and `for_each_value` with types), should a different name be found or maybe the functions constrained?

Reply:
After some discussion, updated to use `template_for_each` for both cases.